src/dom/index.js

```diff
@@ -1,36 +1,63 @@
 import { IS_NON_DIMENSIONAL } from '../constants';
 import options from '../options';
 
+/**
+ * A DOM event listener
+ * @typedef {(e: Event) => void} EventListner
+ */
+
+/**
+ * A mapping of event types to event listeners
+ * @typedef {{ [eventType: string]: EventListener}} EventListenerMap
```

Comment:
I'm ambivalent here. We might find the full-on TS syntax a bit heavy, which could hurt contribution. But, it likely provides better docs (which I think is our end goal here).

Reply:
Sounds like we are learning towards JSDoc syntax so I've updated the types to use JSDoc type syntax where possible (though I could use an extra set of eyes to make sure I didn't miss anything).

3 places where I think we should (or have to in some cases) use TS syntax:
1. (should) Use TS function syntax (e.g. `(arg: string) => string`) in some places so we can specify the parameter types
2. (must) Use TS type operations (namely `&` and `|`) because I don't know of a JSDoc equivalent
3. (must) Use TS 2.9 import type syntax (e.g. `import('./vnode').VNode`) because I don't know of a JSDoc equivalent